cmake/Platform/Common/MSVC/Configurations_msvc.cmake

```diff
@@ -9,10 +9,6 @@
 include(cmake/Platform/Common/Configurations_common.cmake)
 include(cmake/Platform/Common/VisualStudio_common.cmake)
 
-if(NOT CMAKE_GENERATOR MATCHES "Visual Studio 1[6-7]")
-    message(FATAL_ERROR "Generator ${CMAKE_GENERATOR} not supported")
-endif()
```

Comment:
Can we instead do this?
```
if(CMAKE_GENERATOR MATCHES "Visual Studio" AND NOT CMAKE_GENERATOR MATCHES "Visual Studio 1[6-7]")
    message(FATAL_ERROR "Generator ${CMAKE_GENERATOR} not supported")
endif()
```
The intention here was to fail if someone was generating with e.g. VS2017 or VS2015

Reply:
Makes sense to me, 
Pretty much we want to make sure the that if an MSVC toolset is used it is at least v142

Reply:
> The intention here was to fail if someone was generating with e.g. VS2017 or VS2015

There is a check for VS2017 further in the file that can probably be removed then?

if(CMAKE_GENERATOR MATCHES "Visual Studio 15")
    # Visual Studio 2017 has problems with [[maybe_unused]] on lambdas. Sadly, there is no different warning, so 4100 has to remain disabled on 2017
    ly_append_configurations_options(
        COMPILATION
            /wd4100
    )
endif()

Reply:
that check there was my fault, totally missed in my change to add VS2022, @burelc-amzn do you mind?

Reply:
Ah, good call, I'll change it.

Reply:
> Pretty much we want to make sure the that if an MSVC toolset is used it is at least v142

Since this is the intent, I changed the code to check that exactly with
```cmake
if(MSVC_TOOLSET_VERSION VERSION_LESS 142)
```